packages/calypso-products/src/plans-utilities.ts

```diff
@@ -41,10 +41,40 @@ export function getTermDuration( term: string ): number | undefined {
 
 export const redirectCheckoutToWpAdmin = (): boolean => !! JETPACK_REDIRECT_CHECKOUT_TO_WPADMIN;
 
-export const is2023PricingGridEnabled = () => {
-	const supportedLocales = [ 'en', 'en-gb', 'es' ];
-	return (
-		isEnabled( 'onboarding/2023-pricing-grid' ) &&
-		supportedLocales.includes( getLocaleSlug() ?? '' )
-	);
+/**
+ * Returns if the 2023 Pricing grid optimized design should be visible or not
+ *
+ */
+export const is2023PricingGridEnabled = (): boolean => {
+	const isFeatureFlagEnabled = (): boolean => isEnabled( 'onboarding/2023-pricing-grid' );
+
+	const isLocaleSupported = (): boolean => {
+		const supportedLocales = [ 'en', 'en-gb', 'es' ];
+		return supportedLocales.includes( getLocaleSlug() ?? '' );
+	};
+
+	const isActivePage = (): boolean => {
```

Comment:
We are adding complexity to this function with string matching, etc. now. Ideally, we should be considering some form of caching/memoization, cause this will get called a lot on re-rerenders. Can we see what the right way of doing this outside of React? If nothing comes up, then we can memorize the edge/uses.

Reply:
Hey @chriskmnds is this something we can pick up immediately after the release?

I did come up with a solution, but we might need to spend some time on the change to make sure we are achieving the desired optimisation.

PR : #73532
Commit : d319f4b9bf65d498b02db39365ed915bfecd79e0

Reply:
This is a good observation @chriskmnds, let me go back to the drawing board with this.

Reply:
I didn't mean for you to roll your own caching heuristic (also one that looks quite so involved). Maybe worth asking in the javascript channel in Slack for ideas first - there may be recommended ways/libs vs rolling your own. 

For something that tends to lead to over-engineering, maybe an alternative way exists that doesn't involve much complexity e.g. splitting this function apart, creating a memorized selector/wrapper/hook over it for cases where that's used in React, etc. (just rough thoughts)

Reply:
> Hey @chriskmnds is this something we can pick up immediately after the release?

yes, of course